specs/ExecuteScript.md

```diff
@@ -0,0 +1,304 @@
+If ExecuteScript is run during the navigation it should be called after ContentLoading event.
+
+# Examples
+## ExecuteScript
+
+## .NET, WinRT
+```c#
+async void InjectScriptIFrameCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    string iframesData = WebViewFrames_ToString();
+    string iframesInfo = "Enter iframe to run the JavaScript code in.\r\nAvailable iframes: " + iframesData;
+    var dialogIFrames = new TextInputDialog(
+        title: "Inject Script Into IFrame",
+        description: iframesInfo,
+        defaultInput: "0");
+    if (dialogIFrames.ShowDialog() == true)
+    {
+        int iframeNumber = -1;
+        try
+        {
+            iframeNumber = Int32.Parse(dialogIFrames.Input.Text);
+        }
+        catch (FormatException)
+        {
+            Console.WriteLine("Can not convert " + dialogIFrames.Input.Text + " to int");
+        }
+        if (iframeNumber >= 0 && iframeNumber < _webViewFrames.Count)
+        {
+            var dialog = new TextInputDialog(
+                title: "Inject Script",
+                description: "Enter some JavaScript to be executed in the context of iframe " + dialogIFrames.Input.Text,
+                defaultInput: "window.getComputedStyle(document.body).backgroundColor");
+            if (dialog.ShowDialog() == true)
+            {
+                string scriptResult = await _webViewFrames[iframeNumber].ExecuteScriptAsync(dialog.Input.Text);
+                MessageBox.Show(this, scriptResult, "Script Result");
+            }
+        }
+    }
+}
+```
+
+## Win32 C++
+```cpp
+void SampleClass::ExecuteScriptInIframe()
+{
+    std::wstring iframesEnterCode =
+            L"Enter the JavaScript code to run in the iframe " + dialogIFrame.input;
+    TextInputDialog dialogScript(
+            m_appWindow->GetMainWindow(), L"Inject Script Into IFrame", L"Enter script code:",
+            iframesEnterCode.c_str(),
+            L"window.getComputedStyle(document.body).backgroundColor");
+    if (dialogScript.confirmed)
+    {
+        wil::com_ptr<ICoreWebView2Frame> frame;
+        CHECK_FAILURE(m_frames[index]->QueryInterface(IID_PPV_ARGS(&frame)));
+        frame->ExecuteScript(
+            dialogScript.input.c_str(),
+            Callback<ICoreWebView2ExecuteScriptCompletedHandler>(
+                [](HRESULT error, PCWSTR result) -> HRESULT {
+                    if (error != S_OK)
+                    {
+                        ShowFailure(error, L"ExecuteScript failed");
+                    }
+                    MessageBox(nullptr, result, L"ExecuteScript Result", MB_OK);
+                    return S_OK;
+                })
+                .Get());
+    }
+}
+```
+
+## Navigation starting
+
+NavigationStarting is raised when the current frame is requesting permission to navigate to a different URI.
+
+## Win32 C++
+```cpp
+    frame->add_NavigationStarting(
+        Callback<ICoreWebView2FrameNavigationStartingEventHandler>(
```

Comment:
We don't need these samples if they just subscribe to the event and then post a message when they're done. Given that the navigation events all exist and these are frame versions of them I think we don't need samples for all of them. Just the one larger sample above that mixes one navigation event and ExecuteScriptAsync is good

Reply:
Removed.